context/AuthContextProvider.tsx

```diff
@@ -0,0 +1,107 @@
+'use client';
+import { createContext, useContext, useEffect, useMemo, useState } from 'react';
+import { account } from '@/api/config';
+import { useRouter } from 'next/navigation';
+
+type UserCredentials = {
+  email: string;
+  password: string;
+};
+
+type AuthContextType = {
+  isSignedIn: boolean;
+  setIsSignedIn: (isSignedIn: boolean) => void;
+  loginAccount: (user: UserCredentials) => Promise<void | Error>;
+  logoutAccount: () => Promise<void>;
+};
+
+export const AuthContext = createContext<AuthContextType | null>(null);
+
+export const AuthContextProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const [isSignedIn, setIsSignedIn] = useState<boolean>(false);
+  const router = useRouter();
+
+  // Check for a current session on component mount
+  useEffect(() => {
+    const checkSession = async () => {
+      if (!isSessionInLocalStorage()) {
+        console.log('No session data available');
+        return;
+      }
+
+      try {
+        await account.getSession('current');
+        console.log('Active session found');
+        setIsSignedIn(true);
+      } catch (error) {
+        console.error('Session validation error:', error);
+        setIsSignedIn(false);
+      }
+    };
+    checkSession();
+  }, []);
+
+  // Authenticate and set session state
+  const loginAccount = async (user: UserCredentials): Promise<void | Error> => {
+    try {
+      await account.createEmailPasswordSession(user.email, user.password);
+      setIsSignedIn(true);
+    } catch (error) {
+      console.error('Login error:', error);
+      return error as Error;
+    }
+  };
+
+  // Log out and clear session state
+  const logoutAccount = async (): Promise<void> => {
+    try {
+      await account.deleteSession('current');
+      setIsSignedIn(false);
+      await router.push('/login');
+    } catch (error) {
+      console.error('Logout error:', error);
+    }
+  };
+
+  // Helper function to validate session data in local storage
+  const isSessionInLocalStorage = (): boolean => {
+    const loadedDataString = localStorage.getItem('cookieFallback');
+
+    if (!loadedDataString || loadedDataString === '[]') {
+      localStorage.removeItem('cookieFallback');
+      return false;
+    }
+
+    return true;
+  };
+
+  // Memoize context values to avoid unnecessary re-renders
+  const contextValue = useMemo(
+    () => ({
+      isSignedIn,
+      setIsSignedIn,
+      loginAccount,
+      logoutAccount,
+    }),
+    [isSignedIn],
+  );
+
+  return (
+    <AuthContext.Provider value={contextValue}>{children}</AuthContext.Provider>
+  );
+};
+
+// Custom hook to access the authentication context
+export function useAuthContext() {
+  const context = useContext(AuthContext);
+  if (!context) {
+    throw new Error(
```

Comment:
This will throw an error in the console, but what is shown to the user? Looks like we need something to catch the error...

Reply:
sure thing boss.  Let me take a look and see what I can come up with.

Reply:
so this will only be caught in development.

This is if someone tries to access the useAuthContext outside of the AuthContextProvider.

Reply:
Why won't it be caught in Production? 
We should create a ticket to create an Error Boundary component.

Reply:
Great question... in order to use the context you have to add:

```js
import {useAuthContext} from AuthContextProvider
```

then in the component you have to pull out what you need from context:

```js
const {isSignedIn} = useContext
```

if you do this in a component that is NOT wrapped in the <AuthContextProvider> in the layout.tsx file then an error will be thrown right then in there.

there will NEVER be a time in production that the error is thrown.  It would be caught during dev and during build time.

I don't think we need an error boundary component.